### PR TITLE
EOS-14914 Splunk and Ceph automation

### PR DESCRIPTION
**Pull request by nileshgovande**
Signed-off-by: Nilesh Govande <nilesh.govande@seagate.com>

### CODE REVIEW

**Comment by nileshgovande** [comment]
@amitwac2608 and @abhilekh268 - Pls review

File: scripts/cephscript/ceph_script.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+export S3_USE_SIGV4=0
+
+# Log Directory
+#ceph_log_dir="/root/logs/ceph-log"
+ceph_log_dir="/root/sachit"
+mkdir -p "$ceph_log_dir"
+
+while read test_name; do
+        # reading each line
+        #echo $test_name
```
Comment (amitwac2608) [inline scripts/cephscript/ceph_script.sh:45]: remove this comment or make its formatting consistent with above comment (space after hash)

File: scripts/cephscript/ceph_script.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+
+CEPH_DIR="/root/s3-tests"
+CEPH_INIPATH="$CEPH_DIR/seagates3conf.ini"
+CEPH_TEST_DATA="/root/cephtestfilev2.txt"
```
Comment (abhilekh268) [inline scripts/cephscript/ceph_script.sh:24]: we must have same set of tests for v2 and v4 rite....it's just difference signature calculation?

File: scripts/cephscript/ceph_script.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
```
Comment (abhilekh268) [inline scripts/cephscript/ceph_script.sh:20]: can we have argument to script if i want to specifically run V2 or v4

File: scripts/splunkscript/splunk_script.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+
```
Comment (abhilekh268) [inline scripts/splunkscript/splunk_script.sh:21]: same as above let's have a argument to script if we want to specifcally run V2 or V4

File: scripts/cephscript/ceph_script.sh (hunk 2 above)
Comment (amitwac2608) [inline scripts/cephscript/ceph_script.sh:24]: can we have variable name as CEPH_TEST_DATA_V2 ?

File: scripts/cephscript/ceph_script.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+while read test_name; do
+        # reading each line
+        #echo $test_name
+
+        $CEPH_DIR/virtualenv/bin/nosetests $test_name >> $ceph_log_dir/cephrunv2.log 2>&1
+
+done < $CEPH_TEST_DATA
+
+sleep 30s
```
Comment (amitwac2608) [inline scripts/cephscript/ceph_script.sh:51]: should we log or print a message that script will go in sleep for 30s, to make sure we do not give impression that script is hung to the user ?

File: scripts/splunkscript/readme.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+﻿# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (amitwac2608) [inline scripts/splunkscript/readme.md:1]: as this is "md" file, the hash characters here would result in setting the copyright lines as HEADING1.

I would suggest to remove the copyright from readme, as is done elsewhere in the repo.

File: scripts/splunkscript/readme.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+## Steps to Configure & run Splunk compliance test.
+
+Pre-requiste:
+
+Splunk s3test should be installed and configured on the system.
+
+Steps:
+
+1. Virtualenv should be installed on the system
+   yum install python-virtualenv
```
Comment (amitwac2608) [inline scripts/splunkscript/readme.md:28]: convert this in blockquote. You can refer other readme files in s3server repo.

File: scripts/splunkscript/readme.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+   export PYTHONHTTPSVERIFY=0
+
+6. Splunk s3 tests uses AWS signature version 2 by default .If software under test uses signature version 4 then an environment variable need to be defined explicitly.
+   
+   export S3_USE_SIGV4=1
+
+
+## Automation Splunks3test script : splunk_script.sh
+
+ splunk_script.sh is located under " "
```
Comment (amitwac2608) [inline scripts/splunkscript/readme.md:64]: I guess you missed to add text inside the texts !!!

**Review by sachitanands — APPROVED** [review]
looks good

File: scripts/splunkscript/readme.md (hunk 7 above)
Comment (sali619) [reply scripts/splunkscript/readme.md:11]: Its already there in blockquote.
https://github.com/Seagate/cortx-s3server/blob/br/shazia/ceph_splunk_automation/scripts/splunkscript/readme.md

File: scripts/cephscript/ceph_script.sh (hunk 2 above)
Comment (sali619) [reply scripts/cephscript/ceph_script.sh]: We have more awsv4 testcases as comared to awsv2 ,hence the diffrence in testcase  number would be there.

**Review by amitwac2608 — APPROVED** [review]

**Pull request merged**